.circleci/config.yml

```diff
@@ -2,213 +2,193 @@ version: 2
 
 jobs:
   build:
-    working_directory: /go/src/gopkg.in/DataDog/dd-trace-go.v1
-    resource_class: xlarge
-
     docker:
-    - image: circleci/golang:1.12
-    - image: cassandra:3.7
-    - image: circleci/mysql:5.7
-      environment:
-        MYSQL_ROOT_PASSWORD: admin
-        MYSQL_PASSWORD: test
-        MYSQL_USER: test
-        MYSQL_DATABASE: test
-    - image: circleci/postgres:9.5
-      environment:
-        POSTGRES_PASSWORD: postgres
-        POSTGRES_USER: postgres
-        POSTGRES_DB: postgres
-    - image: consul:1.6.0
-    - image: redis:3.2
-    - image: elasticsearch:2
-      environment:
-        ES_JAVA_OPTS: "-Xms750m -Xmx750m" # https://github.com/10up/wp-local-docker/issues/6
-    - image: elasticsearch:5
-      environment:
-        ES_JAVA_OPTS: "-Xms750m -Xmx750m" # https://github.com/10up/wp-local-docker/issues/6
-    - image: datadog/docker-dd-agent
-      environment:
-        DD_APM_ENABLED: "true"
-        DD_BIND_HOST: "0.0.0.0"
-        DD_API_KEY: invalid_key_but_this_is_fine
-    - image: circleci/mongo:latest-ram
-    - image: memcached:1.5.9
-    - image: confluentinc/cp-zookeeper:5.0.0
-      environment:
-        ZOOKEEPER_CLIENT_PORT: "2181"
-    - image: confluentinc/cp-kafka:5.0.0
-      environment:
-        KAFKA_ZOOKEEPER_CONNECT: localhost:2181
-        KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://localhost:9092
-        KAFKA_LISTENERS: PLAINTEXT://0.0.0.0:9092
-        KAFKA_CREATE_TOPICS: gotest:1:1
-        KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: "1"
-
+      - image: circleci/golang:1.12
+        environment:
+          GO111MODULE: "on"
+          GOPATH: "/home/circleci/go"
+    working_directory: /home/circleci/go/src/gopkg.in/DataDog/dd-trace-go.v1
     steps:
     - checkout
 
     - run:
-        name: Check milestone
+        name: milestone
```

Comment:
if we are going to have a workflow base approach could we have all lint checks in a separate job?

- build ------ lint
                 |--- test core
                 |--- test contrib

right now we have all the tests waiting for the lint and if lint fails the tests won't run. It would be nice to know if tests passed and only lint failed.

Reply:
Good idea. I pulled out `lint` to build in parallel with `build` since the linter we're using doesn't seem to need the module to be built to run its checks.